File: level1/문자열_내_마음대로_정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;
    vector<pair<char, string>> cache;

    while(!strings.empty()){
        cache.push_back(make_pair(strings.back()[n], strings.back()));
        strings.pop_back();
    }
                    
    sort(cache.begin(), cache.end());
  
    while(!cache.empty()){
        answer.push_back(cache.back().second);
        cache.pop_back();
    }

    reverse(answer.begin(), answer.end()); 
                    
    return answer;
}

File: JavaScript/level1/문자열_내_마음대로_정렬하기.js
function solution(strings, n) {
    var answer = [];
    var temp = [];
    
    for(let i=0; i<strings.length; i++)
        temp.push([strings[i][n], strings[i]]);  

    temp.sort();
  
    for(let i=0; i<temp.length; i++)
        answer.push(temp[i][1]); 

    return answer;
}
